src/parcsr_ls/par_amg_setup.c

```diff
@@ -625,7 +637,7 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
       k = num_functions - 1;
       for (j = indx-1; j > -1; j--)
       {
-         dof_func[j] = k--;
+         hypre_IntArrayData(dof_func)[j] = k--;
```

Comment:
If `dof_func` is on device, this part needs to ported as well?

Reply:
@liruipeng The CPU code here seems a bit over-complicated... the desired results is:
for (i = 0; i < local_size; i++)
   dof_func[i] = (i + first_local_row) % num_functions
Is that correct? Or am I misunderstanding how the dof_func should be set?

Reply:
> @liruipeng The CPU code here seems a bit over-complicated... the desired results is:
> for (i = 0; i < local_size; i++)
> dof_func[i] = (i + first_local_row) % num_functions
> Is that correct? Or am I misunderstanding how the dof_func should be set?

I think it is correct. I also think CPU code is a bit over-complicated. Not sure why it was coded in that way. We may not use that in the GPU implementation. But first of all, is this part needed to port to GPUs? Just want to make sure if I understood the situation correctly. Thanks!

Reply:
I think this is a pretty cheap operation on the CPU, so this would probably be fine as is if you use unified memory. But, this is straightforward to port, so I'll go ahead and do it. 

By the way, in general, when porting to the GPU, do we expect ported features to work without unified memory? If so, then yes, this section of code definitely needs to be ported (I'd expect it to crash in the case where the user configured without unified memory and then did not pass in dof_func).

Reply:
We want hypre to work without unified memory as much as we can. HIP has no UVM support as far as I know. So, we should port this as well. Thanks again @waynemitchell !